matter_server/server/client_handler.py

```diff
@@ -91,13 +92,17 @@ async def handle_client(self) -> web.WebSocketResponse:
             while not wsock.closed:
                 msg = await wsock.receive()
 
-                if msg.type in (WSMsgType.CLOSE, WSMsgType.CLOSING):
+                if msg.type in (WSMsgType.CLOSED, WSMsgType.CLOSE, WSMsgType.CLOSING):
                     break
 
-                if msg.type != WSMsgType.TEXT:
-                    disconnect_warn = "Received non-Text message."
+                if msg.type == WSMsgType.ERROR:
+                    disconnect_warn = f"Received error message: {msg.data}"
                     break
 
+                if msg.type != WSMsgType.TEXT:
+                    self._logger.debug("Received non-Text message: %s", msg.data)
```

Comment:
Probably the case at hand was a `WSMsgType.CLOSED` message. Since we only support text messages, this really is unexpected, so I'd go for warning:

```suggestion
                    self._logger.warning("Received non-Text message: %s", msg.data)
```